### PR TITLE
IP Addresses: publish IP Addresses for Upbound Infrastructure

### PR DESCRIPTION
**Pull request by darkmuggle**
Add page for Upbound infrastructure IP addresses.

### CODE REVIEW

File: content/reference/ipaddresses.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+title: IP Addresses
+weight: 199
+---
+
+<!-- vale off -->
+## IP Addresses
+
+Upbound Infrastructure uses the following egress addresses for hosted control
```
Comment (AlainRoy) [inline content/reference/ipaddresses.md:9]: These are just the MCP egresses? What about from the main cluster for Librarian & the GitHub App?

I think we might need a slightly more detailed paragraph here. I'm running short on time, but I'll make a suggestion later if I have time. But I think that a reader who doesn't have all your context may have trouble understanding.

Comment (AlainRoy) [reply content/reference/ipaddresses.md:9]: Suggested text:

## Allowing access from Upbound 
Upbound makes outbound connections in three ways:
1. Connections to GitHub to copy the contents of a repository containing your Configuration
2. Connections to GitHub to create your Configurations repository with the Upbound GitHub App. 
3. Connections from Upbound Control Planes to your infrastructure. For example, if you use provider-kubernetes to configure a Kubernetes cluster, there will be a connection to your Kubernetes cluster. 

If you restrict the IP addresses that are allowed to communicate with GitHub or your infrastructure, you'll need to allow the Upbound IP addresses.

To allow access to Github, allow access to the following IP addresses:
* X.X.X.X
* Y.Y.Y.Y

To allow access to your infrastructure from a control plane, allow access to the following IP addresses:
* A.A.A.A
* B.B.B.B

File: content/reference/ipaddresses.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+planes:
+- 35.238.181.48
+- 35.185.250.95
+- 34.72.41.44 
+- 35.238.114.228 
+
+This list of IP Addresses is stable, however, Upbound reserves the right to change this
+list as infrastructure needs change.
```
Comment (plumbis) [inline content/reference/ipaddresses.md:17]:
```suggestion
{{< hint "note" >}}
Upbound reserves the right to change this
list of IP addresses.
{{< /hint >}}
```

File: content/reference/ipaddresses.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+title: IP Addresses
+weight: 199
+---
+
+<!-- vale off -->
+## IP Addresses
```
Comment (plumbis) [inline content/reference/ipaddresses.md:7]:
```suggestion
```

File: content/reference/ipaddresses.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+## Allow List
+
+If you or your organization wish to restrict IP addresses that can acccess APIs, you will
+need to add the above list to your cloud provider(s) and Github. 
+
+Please consult the official documentation of your Cloud Provider:
+- [AWS](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_deny-ip.html)
+- [Azure](https://learn.microsoft.com/en-us/azure/app-service/app-service-ip-restrictions?tabs=azurecli)
+- [GCP](https://cloud.google.com/vpc-service-controls/docs/set-up-private-connectivity)
+- [Github](https://docs.github.com/en/enterprise-cloud@latest/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/managing-allowed-ip-addresses-for-your-organization)
```
Comment (plumbis) [inline content/reference/ipaddresses.md:28]:
```suggestion
```

I'd remove this block. If the user needs the IP list they likely already know how to whitelist the addresses.

**Review by plumbis — CHANGES_REQUESTED** [review]

**Review by plumbis — APPROVED** [review]

**Pull request merged**